src/Parameters/PrefetchDataParameter.php

```diff
@@ -52,24 +52,31 @@ public function resolve(object|null $source, array $args, mixed $context, Resolv
         // So we record all of these ->resolve() calls, collect them together and when a value is actually
         // needed, GraphQL calls the callback of Deferred below. That's when we call the prefetch method,
         // already knowing all the requested fields (source-arguments combinations).
-        return new Deferred(function () use ($info, $context, $args, $prefetchBuffer) {
-            if (! $prefetchBuffer->hasResult($args, $info)) {
-                $prefetchResult = $this->computePrefetch($args, $context, $info, $prefetchBuffer);
-
-                $prefetchBuffer->storeResult($prefetchResult, $args, $info);
+        return new Deferred(function () use ($source, $info, $context, $args, $prefetchBuffer) {
+            if (! $prefetchBuffer->hasResult($source)) {
+                $this->computePrefetch($args, $context, $info, $prefetchBuffer);
             }
 
-            return $prefetchResult ?? $prefetchBuffer->getResult($args, $info);
+            $result = $prefetchBuffer->getResult($source);
+            // clear internal storage
+            $prefetchBuffer->purgeResult($source);
+            return $result;
         });
     }
 
     /** @param array<string, mixed> $args */
-    private function computePrefetch(array $args, mixed $context, ResolveInfo $info, PrefetchBuffer $prefetchBuffer): mixed
+    private function computePrefetch(array $args, mixed $context, ResolveInfo $info, PrefetchBuffer $prefetchBuffer): void
```

Comment:
If this is storing the results, maybe "compute" isn't the right term to use for this method name.  Also "compute" isn't really a great term to be using here.

Reply:
This method was not named by me)
Should I rename it?

doPrefetch | prefetchValues | prefetchResults | processPrefetch | ... ?

Reply:
Renaming a method where the logic contained within it has materially changed, is the right move.  Otherwise, reading and understanding the code for the next developer becomes more confusing.

Being that it's return type is void, it's not really "prefetching results" - you'd expect a return value on that IMO.  `processPrefetch`, `cachePrefetchResults`, `bufferPrefetch`; maybe something along these lines.

Reply:
Renamed method name to processPrefetch pushed to branch